sample-apps/donet-sample-app/Controllers/MetricEmitter.cs

```diff
@@ -0,0 +1,137 @@
+
+        static long apiRequestSent = 0;
+        static long totalBytesSent = 0;
+        static long totalHeapSize  = 0;
+        static long totalThreadSize = 0;
+        static int cpuUsage = 0;
+
+        // The below API name and status code dimensions are currently shared by all metrics observer in
+        // this class.
+        string apiNameValue = "";
+        string statusCodeValue = "";
+
+        
+        public MetricEmitter()
+        {
+            Meter meter = new Meter("adot", "1.0");
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter("adot")
+                .AddOtlpExporter()
+                .Build();
+            
+            string latencyMetricName = API_LATENCY_METRIC;
+            string totalApiRequestSent = API_COUNTER_METRIC;
+            string totalApiBytesSentMetricName = API_SUM_METRIC;
+            string totaltimealiveMetricName = API_TOTAL_TIME_METRIC;
+            string totalHeapSizeMetricName = API_TOTAL_HEAP_SIZE;
+            string totalThreadsMetricName = API_TOTAL_THREAD_SIZE;
+            string cpuUsageMetricName = API_CPU_USAGE;
+
+
+            string instanceId = Environment.GetEnvironmentVariable("INSTANCE_ID");
+            if (instanceId != null && !instanceId.Trim().Equals(""))
+            {
+                latencyMetricName = API_LATENCY_METRIC + "_" + instanceId;
+                totalApiRequestSent = API_COUNTER_METRIC + "_" + instanceId;
+                totalApiBytesSentMetricName = API_SUM_METRIC + "_" + instanceId;
+                totaltimealiveMetricName = API_TOTAL_TIME_METRIC + "_" + instanceId;
+                totalHeapSizeMetricName = API_TOTAL_HEAP_SIZE + "_" + instanceId;
+                totalThreadsMetricName = API_TOTAL_THREAD_SIZE + "_" + instanceId;
+                cpuUsageMetricName = API_CPU_USAGE + "_" + instanceId;
+    
+            }
+            
+
+            meter.CreateObservableCounter(totalApiRequestSent,() => apiRequestSent, 
+                "1",
+                "Increments by one every time a sampleapp endpoint is used");
+
+            meter.CreateObservableCounter(totalApiBytesSentMetricName, () => totalBytesSent, 
+                "By",
+                "Keeps a sum of the total amount of bytes sent while the application is alive");
+
+            apiLatencyRecorder = meter.CreateHistogram<double>(latencyMetricName,
+                 "ms", 
+                 "Measures latency time in buckets of 100 300 and 500");
+
+            totalTimeSentObserver = meter.CreateCounter<long>(totaltimealiveMetricName,
+                "ms",
+                "Measures the total time the application has been alive");
+
+            meter.CreateObservableUpDownCounter(totalHeapSizeMetricName, () => totalHeapSize, 
+                "By",
+                "The current total heap size”");  
+
+            meter.CreateObservableCounter(totalThreadsMetricName, () => totalThreadSize,
```

Comment:
Isn't this supposed to be a synchronous instrument? https://github.com/aws-observability/aws-otel-community/blob/master/sample-apps/SampleAppSpec.md?plain=1#L124

Reply:
Fixed :)